addons/debugger/userscript.js

```diff
@@ -219,32 +239,41 @@ export default async function ({ addon, global, console, msg }) {
     return item;
   };
 
+  let mouseOffsetX = 0;
+  let mouseOffsetY = 0;
+  let lastX = 0;
+  let lastY = 0;
+
   function dragMouseDown(e) {
     e.preventDefault();
-    pos3 = e.clientX;
-    pos4 = e.clientY;
+    mouseOffsetX = e.clientX - consoleWrapper.offsetLeft;
+    mouseOffsetY = e.clientY - consoleWrapper.offsetTop;
+    lastX = e.clientX;
+    lastY = e.clientY;
     document.addEventListener("mouseup", closeDragElement);
     document.addEventListener("mousemove", elementDrag);
   }
 
+  function dragConsole(x, y) {
+    lastX = x;
+    lastY = y;
+    const width = document.documentElement.clientWidth || document.body.clientWidth;
+    const height = document.documentElement.clientHeight || document.body.clientHeight;
+    const clampedX = Math.max(0, Math.min(x - mouseOffsetX, width - consoleWrapper.offsetWidth));
+    const clampedY = Math.max(0, Math.min(y - mouseOffsetY, height - consoleWrapper.offsetHeight));
+    consoleWrapper.style.left = clampedX + "px";
+    consoleWrapper.style.top = clampedY + "px";
+  }
+
```

Comment:
This is letting me drag the logs window exactly 1px more to the bottom that I should, resulting in a vertical scrollbar in the editor